### PR TITLE
Issues/#170 ft 18 sha3 fixes

### PR DESCRIPTION
**Pull request by ealefad**
This PR has fixes for 4 functional tests:
```
rpc_signrawtransaction
p2p_segwit
feature_segwit
feature_block
```
All FT are passing.

### CODE REVIEW

**Comment by ealefad** [comment]
Requires new fixes for `feature_segwit` and `rpc_signrawtransaction` since branch `issues/#18-sha3` has been updated.

**Comment by ealefad** [comment]
All FT fixed, no disabled FT tests.

**Review by r-soltani — APPROVED** [review]

File: eqbtest/functional/feature_block.py [hunk 1]
```diff
@@ -510,7 +509,7 @@ def update_block(block_number, new_transactions):
 
         # Build the redeem script, hash it, use hash to create the p2sh script
         redeem_script = CScript([self.coinbase_pubkey] + [OP_2DUP, OP_CHECKSIGVERIFY]*5 + [OP_CHECKSIG])
-        redeem_script_hash = hash160(redeem_script)
+        redeem_script_hash = hash160_sha3_256(redeem_script)
```
Comment (macterra) [inline eqbtest/functional/feature_block.py:512]: FWIW we call this function SHA3Hash160 on the C++ side

**Review by macterra — APPROVED** [review]

File: eqbtest/functional/feature_block.py (hunk 1 above)
Comment (ealefad) [reply eqbtest/functional/feature_block.py:512]: ah, good point, I didn't check it

**Pull request merged**